lib/telemetry/telemetry.go

```diff
@@ -0,0 +1,90 @@
+package telemetry
+
+// PLSREVIEW(kit): Should this package be lib/telemetry, or should we define it at the top level alongside agent
+//  and lib?
+
+import (
+	"time"
+
+	"github.com/armon/go-metrics"
+)
+
+type Label struct {
+	Key   string
+	Value string
+}
+
+// PLSREVIEW(kit): I decided to directly wrap the API of go-metrics because it informs all of our call sites.
+//  However, this does not need to be set in stone. For example, I renamed Incr to Inc. This is a good opportunity to
+//  adjust the metrics API we use internally. Any suggestions for improvements would be welcome!
+type MetricsClient interface {
+	// A Gauge should retain the last value it is set to
+	SetGauge(key []string, val float32, labels ...Label)
+
+	// KV should emit a Key/Value pair for each call.
+	EmitKV(key []string, val float32, labels ...Label)
+
+	// Counters accumulate
+	IncCounter(key []string, val float32, labels ...Label)
+
+	// Samples are for timing quantiles
+	AddSample(key []string, val float32, labels ...Label)
+	// Convenience fns to capture durations for samples
+	MeasureSince(key []string, start time.Time, labels ...Label)
+
+	// TODO(kit): I have not thought about filters yet. If we don't update them outside of RuntimeCfg
+	//  we don't need to offer an interface.
+
+	// TODO(kit): Should we have functions for rendering out the contents of the Inmemsink or arbitrary metrics state?
+	// 	That would allow us to use everything through a client, rather than having to keep a reference to and
```

Comment:
I think exposing the underlying sink a `GetInmemSink` is a good approach. We don't want to wrap more than we need to.